x-pack/plugins/stack_connectors/server/connector_types/cases/jira/service.ts

```diff
@@ -136,6 +136,9 @@ export const createExternalService = (
     }, '');
   };
 
+  const escapeSpecialCharacters = (str: string) =>
+    str.replace(/[!^&*()+\-[\]\\/{}|:?~]/g, '\\\\$&')?.replace(/-/g, '\\\\x2d');
```

Comment:
I think it is better if we put `/[!^&*()+\-[\]\\/{}|:?~]/g`, `\\\\$&`, etc to constants variables. Why is `replace(/-/g, '\\\\x2d');` needed?

Reply:
Previously it was getting out of range error for RegEx, so had to remove `-`, but I was able to add it to the same character group later. So `replace(/-/g, '\\\\x2d');` is not needed anymore.  Will remove it.

Reply:
fixed in [a73197](https://github.com/elastic/kibana/pull/145610/commits/a73197ef9e85657180c33cd6cee4290f9b8e655a)